apache2/apache2_util.c

```diff
@@ -344,6 +344,14 @@ static void send_waf_log(const char* timestamp, struct waf_lock* lock, apr_file_
     get_short_filename(waf_filename);
     get_ruleset_type_version(waf_ruleset_info, waf_ruleset_type, waf_ruleset_version); 
 
+    // Format UTC time
+    time_t rawtime;
+    struct tm * timeinfo;
+    char timestamp[30];
+    time(&rawtime);
+    timeinfo = gmtime(&rawtime);
```

Comment:
Unfortunately, `gmtime()` is not thread-safe. It stores the resulting structure internally but according to POSIX is not guaranteed to use thread-local storage for that: https://lgtm.com/rules/2154840805/ 

A safe `gmtime_r()` alternative is recommended instead.

Reply:
ah, didn't realize. Thanks.